src/write.c

```diff
@@ -2195,8 +2195,8 @@ static avifResult avifEncoderWriteMediaDataBox(avifEncoder * encoder,
     const size_t mdatStartOffset = avifRWStreamOffset(s);
     for (uint32_t itemPasses = 0; itemPasses < 3; ++itemPasses) {
         // Use multiple passes to pack in the following order:
-        //   * Pass 0: metadata (Exif/XMP)
-        //   * Pass 1: alpha, gain map (AV1)
+        //   * Pass 0: metadata (Exif/XMP/gain map metadata)
```

Comment:
Unrelated to this PR: Shouldn't Exif/XMP be at the end?

Reply:
It's the same as the gain map metadata: if we put it at the end of the file, avifDecoderParse() will block untill it receives the whole file, unless `ignoreExif` and `ignoreXMP` are set. In the documentation for those fields we say `some of these payloads are (unfortunately) packed at the end of the file` so we consider this to be a bad thing.